src/main/java/com/google/sps/servlets/UpdateServlet.java

```diff
@@ -47,10 +48,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       }
     }
     final String updatesInJson = getJsonFromMap(updatesToMake);
-    final String relevantDeviceIds = (String) request.getParameter(DEVICE_IDS_PARAMETER_NAME);
+    final String relevantDeviceIds = request.getParameter(DEVICE_IDS_PARAMETER_NAME);
     final List<String> deviceIds = getDeviceIds(relevantDeviceIds);
     if (!deviceIds.isEmpty() && !updatesToMake.isEmpty()) {
-      utilObj.updateDevices(userId, deviceIds, updatesInJson);
+      List<String> failedUpdateDeviceIds = utilObj.updateDevices(userId, deviceIds, updatesInJson);
+      if (!failedUpdateDeviceIds.isEmpty()) {
+        response.setContentType("application/json");
+        final String json = GSON_OBJECT.toJson(failedUpdateDeviceIds);
+        response.getWriter().println(json);
+      }
     }
     response.sendRedirect(INDEX_URL);
     return;
```

Comment:
It would be helpful if we returned status codes in each case so that the client can easily discern whats going on.  I propose status code 200 for a successful request where all devices have been updated, and maybe status code 500 for situations when some or all of the devices could not be updated.

Reply:
ok